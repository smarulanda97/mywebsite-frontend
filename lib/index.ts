export * from "./drupal.lib";
export * from "./jest.lib";
export * from "./react-query.lib";
