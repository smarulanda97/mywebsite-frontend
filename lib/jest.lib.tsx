import React from "react";
import { ThemeProvider } from "styled-components";
import { QueryClient, QueryClientProvider } from "react-query";
import { render as defaultRender, RenderResult } from "@testing-library/react";

import { dark } from "@/styles";

type DefaultParameters = Parameters<typeof defaultRender>;
type RenderUI = DefaultParameters[0];
type RenderOptions = DefaultParameters[1];

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});

export const Wrapper: React.FC = ({ children }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={dark}>{children}</ThemeProvider>
    </QueryClientProvider>
  );
};

export function render(ui: RenderUI, { wrapper, ...options }: RenderOptions = {}): RenderResult {
  if (!wrapper) {
    // eslint-disable-next-line react/display-name
    wrapper = Wrapper;
  }

  return defaultRender(ui, { wrapper, ...options });
}
