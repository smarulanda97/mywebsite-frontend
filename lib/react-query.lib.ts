import { QueryClient as ReactQueryClient } from "react-query";

export const queryClient = new ReactQueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});
