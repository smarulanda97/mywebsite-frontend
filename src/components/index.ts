export * from "./Node/Node.component";
export * from "./Layout/Layout.component";
export * from "./Paragraph/Paragraph.component";
export * from "./PreviewAlert/PreviewAlert.component";

// UI components
export * from "./Toolkit/Toolkit.component";
