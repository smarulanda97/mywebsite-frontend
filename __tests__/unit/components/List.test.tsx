import { render } from "@/lib";
import { List } from "@/components";

describe("<List>", function () {
  const placeholders = [
    {
      id: 1,
      text: "hello",
    },
    {
      id: 2,
      text: "world",
    },
  ];

  test("Renders correctly", function () {
    const component = render(
      <List
        title="placeholder list"
        elements={placeholders}
        render={(element) => <span>{element.text}</span>}
      />
    );

    // Checks styles
    const list = component.getByRole("list");
    expect(list).toBeInTheDocument();
    expect(list).toHaveStyle({ display: 'flex' });

    // Checks elements rendered
    expect(component.container).toHaveTextContent("placeholder list");
    expect(component.getAllByRole("listitem")).toHaveLength(2);
    expect(component.container).toHaveTextContent("world");
    expect(component.container).toHaveTextContent("hello");
  });

  test("Title prop could be optional", function () {
    const { container } = render(
      <List elements={placeholders} render={(element) => <span>{element.text}</span>} />
    );

    expect(container).not.toHaveTextContent("placeholder list");
  });

  test("UI shouldn't change", function () {
    const { container } = render(
      <List
        title="placeholder list"
        elements={placeholders}
        render={(element) => <span>{element.text}</span>}
      />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
