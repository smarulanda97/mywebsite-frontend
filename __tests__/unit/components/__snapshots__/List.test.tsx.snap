// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List> UI shouldn't change 1`] = `
.c0 {
  font-size: 14px;
  text-transform: uppercase;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-weight: 500;
  display: block;
  color: #ffffff;
  margin-bottom: 20px;
}

.c1 {
  margin: -10px -12px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  list-style: none;
}

.c1 li {
  margin: 10px 12px;
}

.c1 a {
  display: block;
  width: 60px;
  height: 60px;
  line-height: 60px;
  text-align: center;
  font-size: 14px;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  border-radius: 6px;
}

.c1 a > span {
  vertical-align: middle;
}

.c1 a > span img {
  height: 32px !important;
}

@media (max-width:575px) {
  .c0 {
    text-align: center;
  }
}

@media (max-width:575px) {
  .c1 {
    margin: 0 auto;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:575px) {
  .c1 li:first-child {
    margin-left: 0;
  }
}

<div
  class=""
>
  <span
    class="c0"
  >
    placeholder list
  </span>
  <ul
    class="c1"
  >
    <li>
      <span>
        hello
      </span>
    </li>
    <li>
      <span>
        world
      </span>
    </li>
  </ul>
</div>
`;
